src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ExpressionTreeCallRewriter.cs

```diff
@@ -74,14 +74,10 @@ protected override Expr VisitSAVE(ExprBinOp pExpr)
         {
             // Saves should have a LHS that is a CALL to PM_EXPRESSION_PARAMETER
             // and a RHS that is a WRAP of that call.
-            Debug.Assert(pExpr.OptionalLeftChild != null);
-            Debug.Assert(pExpr.OptionalLeftChild.isCALL());
-            Debug.Assert(pExpr.OptionalLeftChild.asCALL().PredefinedMethod == PREDEFMETH.PM_EXPRESSION_PARAMETER);
-            Debug.Assert(pExpr.OptionalRightChild != null);
-            Debug.Assert(pExpr.OptionalRightChild.isWRAP());
-
-            ExprCall call = pExpr.OptionalLeftChild.asCALL();
-            ExprTypeOf TypeOf = call.OptionalArguments.asLIST().OptionalElement.asTYPEOF();
+            ExprCall call = pExpr.OptionalLeftChild as ExprCall;
```

Comment:
It can't be null. If it was then the assertion would fail the result of `?.PredefinedMethod` would not be equal to `PREDEFMETH.PM_EXPRESSION_PARAMETER`, so it serves to assert both non-nullity and the value of that property.
(There's something to be said for not asserting too much in one expression, but there's something to be said for not having several lines of assertions that begin to blend into each other on reading, too).